src/sink/kv.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: Add correlation of reads and replies.
+
+#![cfg(not(tarpaulin_include))]
+use crate::sink::{prelude::*, Reply};
+use crate::source::prelude::*;
+use async_channel::Sender;
+use halfbrown::HashMap;
+use serde::Deserialize;
+use sled::{CompareAndSwapError, IVec};
+use std::boxed::Box;
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct Config {
+    dir: String,
```

Comment:
what about a config like in the `WAL` operator `flush_on_event` that will flush the kv store on every event?

Reply:
or an explicit `flush` command?

Reply:
I wanted to avoid that complexity as the requirements are not that of a WAL, I fear that will add more confusion and misuse than just giving one default option. We want this to be accessible to people without making them read on the details of the behavior of storage engine flushes, how their particular OS is working and what the effect between those will be.